### PR TITLE
GRAPHICS: MACGUI: Make use of Common::String::format

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

File: graphics/macgui/macfontmanager.cpp [hunk 1]
```diff
@@ -327,12 +331,10 @@ const char *MacFontManager::getFontName(int id, int size, int slant, bool tryGen
 		slant = 0;
 	}
 
-	snprintf(name, 128, "%s-%d-%d", n.c_str(), slant, size);
-
-	return name;
+	return Common::String::format("%s-%d-%d", n.c_str(), slant, size);
```
Comment (bluegr) [inline graphics/macgui/macfontmanager.cpp]: A Common::String variable is still needed here, as Common::String::format() is a static method, so no new instance of a Common::String is created. You could do:

    return Common::String(Common::String::format(...));

File: graphics/macgui/macmenu.cpp [hunk 2]
```diff
@@ -391,14 +391,11 @@ const Font *MacMenu::getMenuFont() {
...
-	*res = 0;
+const Common::String MacMenu::getAcceleratorString(MacMenuSubItem *item, const char *prefix) {
+	if (item->shortcut == 0)
+		return Common::String();
 
-	if (item->shortcut != 0)
-		sprintf(res, "%s%c%c", prefix, (_wm->_fontMan->hasBuiltInFonts() ? '^' : '\x11'), item->shortcut);
-
-	return res;
+	return Common::String::format("%s%c%c", prefix, (_wm->_fontMan->hasBuiltInFonts() ? '^' : '\x11'), item->shortcut);
```
Comment (bluegr) [inline graphics/macgui/macmenu.cpp]: Same here. A Common::String variable is still needed here, as Common::String::format() is a static method, so no new instance of a Common::String is created. You could do:

    return Common::String(Common::String::format(...));

File: graphics/macgui/macfontmanager.cpp (hunk 1 above)
Comment (criezy) [reply graphics/macgui/macfontmanager.cpp]: This code seems fine to me since Common::String::format() returns a Common::String() already.

Comment (bluegr) [reply graphics/macgui/macfontmanager.cpp]: True :)

**Comment by bluegr** [comment]
Nice work, thanks! Merging

**Pull request merged**